propertyestimator/layers/reweighting.py

```diff
@@ -42,24 +90,188 @@ def schedule_calculation(calculation_backend, storage_backend, layer_directory,
-        existing_data: list of StoredSimulationData
-            Data which has been stored from previous calculations on systems
-            of the same composition as the desired property.
+        parameter_set_id: :obj:`str`
+            The id of the force field parameters which the property should be
+            estimated with.
+        existing_data: :obj:`dict` of :obj:`str` and :obj:`str`
+            A file path to data which has been stored from previous calculations on
+            systems of the same composition as the desired property.
+        existing_force_fields: :obj:`dict` of :obj:`str` and :obj:`str`
+            A path to a dictionary of all of the force field parameters referenced by the
+            `existing_data`, which have been serialized with `serialize_force_field`
+        """
+        property_class = physical_property.__class__
+
+        if not hasattr(property_class, 'reweight'):
+            return None
+
+        existing_data = {}
+        existing_force_fields = {}
+
+        for substance_id in existing_data_paths:
+
+            existing_data[substance_id] = []
+
+            for data_path in existing_data_paths[substance_id]:
+
+                with open(data_path, 'rb') as file:
+                    existing_data[substance_id].append(pickle.load(file))
+
+        for force_field_id in existing_force_field_paths:
+
+            with open(existing_force_field_paths[force_field_id], 'rb') as file:
+                existing_force_fields[force_field_id] = deserialize_force_field(pickle.load(file))
+
+        reweighted_property = property_class.reweight(physical_property, options, parameter_set_id, existing_data,
+                                                      existing_force_fields, available_resources)
+
+        return_object = CalculationLayerResult()
+        return_object.property_id = physical_property.id
+
+        return_object.calculated_property = reweighted_property
+        return_object.data_to_store = []
+
+        return return_object
+
+    @staticmethod
+    def get_reduced_potential(substance, thermodynamic_state, target_force_field_id,
+                              target_force_field, existing_data, available_resources):
+        """Get the reduced potential from an existing data set. If the target force field
+        id does not match the force field id used by the existing data, the reduced potential
+        will be re-evaluated using the target force field.
+
+        Parameters
+        ----------
+        thermodynamic_state: The thermodynamic state to use when reducing the
+            energies.
+        target_force_field_id: str
+            The id of the target force field parameters.
+        target_force_field: ForceField
+            The target set of force field parameters.
+        existing_data: StoredSimulationData
+            The reference data to get the energies from.
+
+        Returns
+        -------
+        :obj:`numpy.ndarray`
```

Comment:
What dimensionality is the array? Is it something like `(nsnapshots)` in dimension?

Reply:
It is - I've added this to the docstring.